pkg/virt-controller/watch/vm/vm.go

```diff
@@ -3254,3 +3260,184 @@ func (c *Controller) handleMemoryHotplugRequest(vm *virtv1.VirtualMachine, vmi *
 
 	return nil
 }
+
+func hasEphemeralHotplugVolumes(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineInstance) bool {
```

Comment:
I am not sure I understand this function. We are checking the VMI to see if there are hotplugged volumes in there, and then returning true. And if I understand correctly true means the VMI has hotplugged volumes while the VM does not. But we don't check if the volumes found in the VMI exist in the VM, which would make them hotplugged, but not ephemeral.

Reply:
This is functionality that @alicefr added for `Replacement` and `Migration` `UpdateVolumesStrategy`. Honestly not sure why but I wanted to maintain it. However it does not work with `Hotplug` strategy which is why we avoid it in that case.

To your point about checking the VM, see here: https://github.com/mhenriks/kubevirt/blob/6844567b8c03b57506abb27a0d4b045562b969c7/pkg/virt-controller/watch/vm/vm.go#L3277-L3278

Reply:
Do you need this in order to do a correctly check if the 2 volumes set to not interpret the lack of a volume as a removal from the VM?

Reply:
> Do you need this in order to do a correctly check if the 2 volumes set to not interpret the lack of a volume as a removal from the VM?

I effectively skip this function here: https://github.com/mhenriks/kubevirt/blob/decl-hotplug/pkg/virt-controller/watch/vm/vm.go#L3266-L3268

With hotplug strategy there is code later to check if volumes are added/removed